lib/mobility/ransack.rb

```diff
@@ -1,6 +1,56 @@
 require "mobility/ransack/version"
+require "mobility/plugins/arel"
+require "ransack"
 
 module Mobility
   module Ransack
+    class Visitor < Mobility::Plugins::Arel::Visitor
+      def initialize
+        super(nil, nil)
+      end
+
+      private
+
+      def visit_collection(objects)
+        objects.map(&method(:visit)).sum.uniq
+      end
+      alias :visit_Array :visit_collection
+      alias :visit_Arel_Nodes_Equality :visit_Arel_Nodes_Binary
+      alias :visit_Arel_Nodes_Or :visit_Arel_Nodes_Binary
+
+      def visit_Mobility_Plugins_Arel_Attribute(object)
+        [[object.backend_class, object.locale]]
```

Comment:
This is relying on the undocumented fact that backends which require applying additional scope (KeyValue and Table) define their nodes as objects of class `Mobility::Plugins::Arel::Attribute`. If this were not the case then this would not work.

I think Mobility could have a better way to just say:

```ruby
Mobility.apply_scope do
  < some stuff with backend nodes >
end
```

and it would work. Then we could just use that here. But that would require a documented standard for how to define backend nodes which require applied scopes (i.e. backends that require joins) which we don't currently have.

Anyway this should work for now. Maybe we can add something like this to Mobility 1.2 :thinking: